### PR TITLE
Add a subclassable test

### PR DESCRIPTION
**Pull request by elmarco**
This test was written to illustrate issue #8

### CODE REVIEW

**Review by GuillaumeGomez — DISMISSED** [review]

**Comment by elmarco** [comment]
@GuillaumeGomez it doesn't compile :)

**Comment by GuillaumeGomez** [comment]
Doesn't matter, you have my full approval anyway. :)

**Comment by bilelmoussaoui** [comment]
> @GuillaumeGomez it doesn't compile :)

The CI says otherwise :)

**Comment by sdroege** [comment]
> The CI says otherwise :)

That's also why the repeated pings in https://github.com/gtk-rs/gtk-rs-core/issues/8 @elmarco

**Comment by elmarco** [comment]
CI doesn't cover cargo test?
```
 elmarco@kabiry  ~/src/gtk-rs-core   issue-8  cargo test  
   Compiling glib-macros v0.13.0 (/home/elmarco/src/gtk-rs-core/glib-macros)
error[E0277]: the trait bound `O: glib::IsA<O>` is not satisfied
   --> glib-macros/tests/test.rs:276:58
    |
276 |                 let self_ = imp::Foo::from_instance(self.downcast_ref::<Foo>().unwrap());
    |                                                          ^^^^^^^^^^^^ the trait `glib::IsA<O>` is not implemented for `O`
    |
    = note: required because of the requirements on the impl of `CanDowncast<foo::Foo>` for `O`
help: consider further restricting this bound
    |
274 |         impl<O: IsA<Foo> + glib::IsA<O>> FooExt for O {
    |                          ^^^^^^^^^^^^^^

error[E0277]: the trait bound `foo::Foo: glib::IsA<O>` is not satisfied
   --> glib-macros/tests/test.rs:276:58
    |
276 |                 let self_ = imp::Foo::from_instance(self.downcast_ref::<Foo>().unwrap());
    |                                                          ^^^^^^^^^^^^ the trait `glib::IsA<O>` is not implemented for `foo::Foo`
    |
    = note: required because of the requirements on the impl of `CanDowncast<foo::Foo>` for `O`
help: consider introducing a `where` bound, but there might be an alternative better way to express this requirement
    |
274 |         impl<O: IsA<Foo>> FooExt for O where foo::Foo: glib::IsA<O> {
    |                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by bilelmoussaoui** [comment]
Oh, we don't run tests for glib-macros at all...

**Comment by GuillaumeGomez** [comment]
@bilelmoussaoui Want to send a PR to fix that?

**Comment by bilelmoussaoui** [comment]
> @bilelmoussaoui Want to send a PR to fix that?

https://github.com/gtk-rs/gtk-rs-core/pull/82

**Comment by bilelmoussaoui** [comment]
Do you mind rebasing your PR so that the glib-macros tests are run?

File: glib-macros/tests/test.rs [hunk 1]
```diff
@@ -243,3 +243,43 @@ fn attr_gflags() {
...
+            impl ObjectImpl for Foo {}
+        }
+
+        pub trait FooExt: 'static {
+            fn test(&self);
+        }
+
+        impl<O: IsA<Foo>> FooExt for O {
+            fn test(&self) {
+                let self_ = imp::Foo::from_instance(self.downcast_ref::<Foo>().unwrap());
```
Comment (bilelmoussaoui) [inline glib-macros/tests/test.rs:276]:
```suggestion
                let self_ = imp::Foo::from_instance(self.as_ref().downcast_ref::<Foo>().unwrap());
```
it needs this :-)

File: glib-macros/tests/test.rs [hunk 2]
```diff
@@ -243,3 +243,43 @@ fn attr_gflags() {
     assert!(e.value_by_nick("ab").is_none());
     assert!(e.value_by_nick("c").is_some());
 }
+
+#[test]
+fn subclassable() {
+    mod foo {
+        use super::*;
+        use glib::subclass::prelude::*;
+        use glib::subclass::types::ObjectSubclassExt;
```
Comment (bilelmoussaoui) [inline glib-macros/tests/test.rs:252]:
```suggestion
```
already in the prelude

File: glib-macros/tests/test.rs (hunk 1 above)
Comment (elmarco) [reply glib-macros/tests/test.rs]: ah ok!

**Review by bilelmoussaoui — APPROVED** [review]

**Review by sdroege — COMMENTED** [review]
I guess this is still useful to merge for testing the `object_subclass` macro?

**Comment by bilelmoussaoui** [comment]
> I guess this is still useful to merge for testing the `object_subclass` macro?

especially if we end up simplifying things a bit somehow in the future

**Review by sdroege — APPROVED** [review]

**Pull request merged**